pkg/porter/credentials.go

```diff
@@ -146,6 +151,86 @@ func (o *CredentialShowOptions) Validate(args []string) error {
 	return o.ParseFormat()
 }
 
+// Validate validates the args provided to Porter's credential edit command
+func (o *CredentialEditOptions) Validate(args []string) error {
+	if err := validateCredentialName(args); err != nil {
+		return err
+	}
+	o.Name = args[0]
+	return nil
+}
+
+// EditCredential edits the credentials of the provided name.
+func (p *Porter) EditCredential(opts CredentialEditOptions) error {
+	credSet, err := p.Credentials.Read(opts.Name)
+	if err != nil {
+		return err
+	}
+
+	if len(credSet.Credentials) == 0 {
+		return errors.New("no credentials exist in this set")
+	}
+
+	// use a flattened data structure for the credentials, so when we marshal
+	// the structure to yaml it's easier for the user to edit.
+	type tempStrategy struct {
```

Comment:
Is the intent behind the custom types to remove the extra level of indentation? e.g. 

```yaml
source:
  env:...
```

Or is it to not have they use the key value pair with a dynamic key?

```yaml
env: MY_ENV_VAR
```

We could alter the format of the credential set, though anyone who looks at the data on disk or in the cloud would be confused about why it looks different. I'd be okay with that if we had a clear reason on why we needed to hide/alter that for the user in this editing mode.

I am concerned that the words we have chosen may not map well to how people think about the credential based on how they made it, e.g. "type". They aren't different types of credentials, and when we asked them to create it we didnt' use that word. But before bikeshedding, I'd like to make sure we really want to alter the structure from the user in the first place.

Reply:
Primary reason was to simplify the structure, a bit of both of those reasons. But as you say, altering this will be confusing to users, and I totally agree, I was a bit unsure of this in the first place.

Reply:
How about for now let's directly expose the underlying credential set structure and then pay attention to how well that is working for people. If we get feedback that the language (yaml) or the structure is making it easy to make mistakes, we can iterate and provide a simplified structure for them to edit, or consider tweaking the language (indenting is hard, no joke).